### PR TITLE
Add support for animating CATransform3D

### LINKED ISSUE #19: Make CATransform3D conform to AnimatableProperty
`CATransform3D` is a pretty common property type to animate when dealing with layers (it's the type for `CALayer.transform` and `CALayer.sublayerTransform`), so it would be great to have conformance handled by the framework.

### PR DESCRIPTION
**Pull request by NickEntin**
This adds conformance for CATransform3D to AnimatableProperty by decomposing the two matrices into their respective components, interpolating the values in the decomposed transforms, then recomposing the interpolated value into a CATransform3D. This follows the same pattern we use to interpolate CGAffineTransform.

Resolves #19.

### CODE REVIEW

**Comment by NickEntin** [comment]
I also tried a version of this [using simd types](https://github.com/NickEntin/stagehand/commit/d4e47f99bbe0f5e6bef4de9e462f3237008e3f39), but it had worse performance (by ~15%).

File: Sources/Stagehand/Utilities/Vector4.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import CoreGraphics
+import QuartzCore
+
+internal struct Vector4 {
+
+    // MARK: - Public Properties
```
Comment (seanho) [inline Sources/Stagehand/Utilities/Vector4.swift]: Internal?

**Review by seanho — APPROVED** [review]
I do not understand most of the math stuffs in there, just reviewing the change set as is and code looks good. The test snapshots are very useful and cool, one suggestion, if we can make the red block different color on each side, it will make the rotation/perspective snapshot animation more obvious.

**Comment by NickEntin** [comment]
> one suggestion, if we can make the red block different color on each side, it will make the rotation/perspective snapshot animation more obvious.

That sounds like it would be super helpful, but as far as I know CoreAnimation doesn't support this. Layers are really 2D objects in a 3D world, so there is no "back side" to the layer.

File: Sources/Stagehand/Utilities/Vector4.swift (hunk 1 above)
Comment (NickEntin) [reply Sources/Stagehand/Utilities/Vector4.swift]: Yep, good catch!

**Comment by NickEntin** [comment]
> as far as I know CoreAnimation doesn't support this

Chatted offline. I misunderstood what the suggestion was. Commit coming shortly to address this.

**Comment by NickEntin** [comment]
So... the last segment of the [shear interpolation test](https://github.com/cashapp/stagehand/blob/86d55fb5eb96e9bb9e53eabe15f409339ff16b2b/Example/Unit%20Tests/ReferenceImages/_64/Stagehand_UnitTests.CATransform3DInterpolationSnapshotTests/testShear_13_3_375x812%403x.png) looks really weird. But it's actually doing the right thing. I'll replace it with a better example.

**Pull request merged**